eplaunch/interface/frame.py

```diff
@@ -77,17 +78,26 @@ def build_primary_toolbar(self):
         self.tb.SetToolBitmapSize(t_size)
 
         ch_id = wx.NewId()
-        workflow_choices = [
-            "EnergyPlus SI (*.idf, *.imf)",
-            "EnergyPlus IP (*.idf, *.imf)",
-            "AppGPostProcess (*.html)",
-            "CalcSoilSurfTemp (*.epw)",
-            "CoeffCheck (*.cci)",
-            "CoeffConv (*.coi)",
-            "Basement (*.idf)",
-            "Slab (*.idf)",
-            "File Operations (*.*)"
-        ]
+
+        workflow_choices = []
+        built_in_workflow_classes = workflow_manager.get_workflows()
+        for workflow_class in built_in_workflow_classes:
+            workflow_instance = workflow_class()
+            workflow_name = workflow_instance.name()
+            workflow_file_types = workflow_instance.get_file_types()
+
+            file_type_string = "("
+            first = True
+            for file_type in workflow_file_types:
+                if first:
+                    first = False
+                else:
+                    file_type_string += ", "
+                file_type_string += file_type
+            file_type_string += ")"
+
+            workflow_choices.append("%s %s" % (workflow_name, file_type_string))
+
```

Comment:
This is the great change.  Now instead of a hardwired list of workflows, I am actually processing workflow class files that I have defined elsewhere and processing those into a meaningful list of current workflows.  This should be extracted into its own function so that we can easily update the list while the application is open using a button or what not.  I also have to work out where the updated workflow instances live...

Reply:
This is great!